semgrep-core/src/cli/Main.ml

```diff
@@ -402,7 +402,12 @@ let cache_computation file cache_file_of_file f =
           res)
         else
           let res = f () in
-          Common2.write_value (Version.version, file, res) file_cache;
+          (try Common2.write_value (Version.version, file, res) file_cache
+           with Sys_error err ->
```

Comment:
maybe a comment here that talks about SIGXFSZ?
What is the size limit actually? Why do we reach it? Our ASTs are too big?

Reply:
I don't know exactly how this happens but in some shells `ulimit -f` shows me `unlimited` whereas in others shows me `10000` (file blocks). I certainly have not set that limit myself, it's very strange. I get the error on a file whose AST dump has a size of 19,993,228 bytes (so not that large).

Reply:
Just to check, do you get this error always or when `ulimit -f` shows you `10000`?

Reply:
If there is no file size limit then I don't get the error.